src/Reflection/ReflectionClass.php

```diff
@@ -801,15 +838,19 @@ public function getParentClass() : ?ReflectionClass
             return null;
         }
 
-        $parent = $this->reflector->reflect($this->node->extends->toString());
-        // @TODO use actual `ClassReflector` or `FunctionReflector`?
-        assert($parent instanceof self);
+        if ($this->cachedParentClass === null) {
+            $parent = $this->reflector->reflect($this->node->extends->toString());
+            // @TODO use actual `ClassReflector` or `FunctionReflector`?
```

Comment:
I think this TODO is solved for now: we can improve on the type system of reflectors later on.

Reply:
Not relevant for this patch: can be removed later